### PR TITLE
feat: (MSP-11) Initiate git-terminator scripts

### PR DESCRIPTION
**Pull request by javiercm1410**

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
Comment (angelmadames) [inline .husky/commit-msg:2]: Not needed in the most recent versions of husky.

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1 @@
+bun 1.1.21
```
Comment (angelmadames) [inline .tool-versions:1]: We should add Renovate to this repo.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
+{
+  "name": "terminator",
```
Comment (angelmadames) [inline package.json:2]: We should have the full repository name in here.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    "check:fix": "biome check src/ --apply",
+    "test": "bun test",
+    "prepare": "husky"
+  },
+  "devDependencies": {
+    "@biomejs/biome": "^1.8.3",
+    "@commitlint/config-conventional": "^19.2.2",
+    "@release-it/conventional-changelog": "^8.0.1",
+    "@types/bun": "latest",
+    "commander": "^12.1.0",
```
Comment (angelmadames) [inline package.json:28]: Commander is a production dependency.

File: src/utils/delete_tags.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
+import type { Octokit } from "@octokit/core";
```
Comment (angelmadames) [inline src/utils/delete_tags.ts:1]: In the TypeScript/JavaScript world, file names are `kebak-case` or `camelCase`. `snake_case` is ideal for Python, Go, Ruby, Elixir and others.

File: src/utils/delete_tags.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+			);
+		}
+	} catch (error) {
+		console.error("Eploto:", error);
+		process.exit(1);
+	}
+};
+
+// This func could be added as methods of my githubConfig class
+// Need test
```
Comment (angelmadames) [inline src/utils/delete_tags.ts:63]: Avoid unnecessary comments (prefer using `@TODO:` mark).

File: src/utils/delete_tags.ts [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+		const octokit = githubConfig.auth();
+		let repoList = repositories;
+
+		if (repositories.length < 1) {
+			const orgRepositories = await octokit.request("GET /orgs/{org}/repos", {
+				org: org,
+				headers: {
+					"X-GitHub-Api-Version": "2022-11-28",
+				},
+			});
```
Comment (angelmadames) [inline src/utils/delete_tags.ts:37]: I know that this appears in the official documentation of GitHub API, but I actually don't like using `.request`. I prefer using `.rest.<method>` for these kind of things. It's more readable and allows you to compose the configuration of all requests elsewhere (ie. defining headerS).

Check these references:
- https://github.com/octokit/octokit.js?tab=readme-ov-file#octokit-api-client
- https://docs.github.com/en/rest/guides/scripting-with-the-rest-api-and-javascript?apiVersion=2022-11-28#using-rest-endpoint-methods-to-make-requests

File: src/utils/delete_tags.ts [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+					//   await octokit.request('DELETE /repos/{owner}/{repo}/git/refs/{ref}', {
+					//     owner: gitRepo.name,
+					//     repo: gitRepo.name,
+					//     ref: 'tags/'+tag.name,
+					//     headers: {
+					//       'X-GitHub-Api-Version': '2022-11-28'
+					//     }
+					//   });
+					//   console.log("%s Tag deleted!", tag.name)
+					// }
```
Comment (angelmadames) [inline src/utils/delete_tags.ts:143]: Avoid commented code.

File: src/utils/delete_tags.ts [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+	dry: boolean,
+) => {
+	try {
+		tagsDate.splice(0, minTags);
+
+		for (const tag of tagsDate) {
+			const tagDaysSinceCreated =
+				(new Date().getTime() - new Date(tag.date).getTime()) / 86400000;
+			if (tagDaysSinceCreated > daysUntilStale) {
+				// These console logs are for Testing
```
Comment (angelmadames) [inline src/utils/delete_tags.ts:127]: These should live elsewhere. Prefer using tests.

**Review by angelmadames — CHANGES_REQUESTED** [review]

File: .husky/commit-msg (hunk 1 above)
Comment (javiercm1410) [reply .husky/commit-msg:2]: Removed!

File: .tool-versions (hunk 2 above)
Comment (javiercm1410) [reply .tool-versions:1]: Installed and configuration added

File: package.json (hunk 4 above)
Comment (javiercm1410) [reply package.json:28]: installed

File: src/utils/delete_tags.ts (hunk 5 above)
Comment (javiercm1410) [reply src/utils/delete_tags.ts:1]: filename updated

File: src/utils/delete_tags.ts (hunk 6 above)
Comment (javiercm1410) [reply src/utils/delete_tags.ts:63]: Comments removed

File: package.json (hunk 3 above)
Comment (javiercm1410) [reply package.json]: solved

File: src/utils/delete_tags.ts (hunk 8 above)
Comment (javiercm1410) [reply src/utils/delete_tags.ts]: comments removed

File: src/utils/delete_tags.ts (hunk 9 above)
Comment (javiercm1410) [reply src/utils/delete_tags.ts]: solved

File: command.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    'Time until a tag is considered stale in days',
+    '365'
+  )
+  .requiredOption(
+    '-o, --organization <organization>',
+    'GitHub Organization'
+  )
+  .option(
+    '-r, --repository <repository...>',
+    'Minimum number of Tags to preserve',
```
Comment (javiercm1410) [inline command.ts:24]: Change option description

File: src/utils/delete_tags.ts (hunk 7 above)
Comment (javiercm1410) [reply src/utils/delete_tags.ts]: updated!

**Pull request merged**